tests/migrations/test_autogen_migration.py

```diff
@@ -0,0 +1,160 @@
+import pytest
+
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations.autogeneration.autogen_migrations import generate_migration_ops
+from snuba.migrations.columns import MigrationModifiers
+from snuba.migrations.operations import AddColumn, DropColumn, OperationTarget
+from snuba.utils.schemas import Column, DateTime, UInt
+
+
+def mockstoragewithcolumns(cols: list[str]) -> str:
+    colstr = ",\n            ".join([s for s in cols])
+    return f"""
+version: v1
+kind: writable_storage
+name: errors
+storage:
+    key: errors
+    set_key: events
+readiness_state: complete
+schema:
+    columns:
+        [
+            {colstr}
+        ]
+    local_table_name: errors_local
+    dist_table_name: errors_dist
+    partition_format:
+        - retention_days
+        - date
+    not_deleted_mandatory_condition: deleted
+local_table_name: errors_local
+dist_table_name: errors_dist
+"""
+
+
+def test_add_column() -> None:
+    cols = [
+        "{ name: project_id, type: UInt, args: { size: 64 } }",
+        "{ name: timestamp, type: DateTime }",
+        "{ name: event_id, type: UUID }",
+    ]
+    new_cols = [
+        "{ name: project_id, type: UInt, args: { size: 64 } }",
+        "{ name: timestamp, type: DateTime }",
+        "{ name: newcol1, type: DateTime }",
+        "{ name: event_id, type: UUID }",
+        "{ name: newcol2, type: UInt, args: { schema_modifiers: [nullable], size: 8 } }",
```

Comment:
what happens if the newcol is not appended at the end of the columns array?

Reply:
it will still work (see newcol1), i just tested it but I can add a test where 1 isnt at the end if you think it would be good.